app/Controllers/Graph.hs

```diff
@@ -38,9 +37,10 @@ graphResponse =
 
 
 index :: ServerPart Response
-index = liftIO (runSqlite databasePath $ do
-    graphsList :: [Entity Graph] <- selectList [GraphDynamic ==. False] [Asc GraphTitle]
-    return $ createJSONResponse graphsList :: SqlPersistM Response)
+index = do
```

Comment:
There's an extra `do` here.

If you're having trouble with types and compilation, there's a deeper problem to investigate (`do` is not the answer). This is true throughout.